### PR TITLE
Fix memory leak (start enum)

### PR DESCRIPTION
**Pull request by cedric-chaumont-st-dev**
Signed-off-by: Cedric Chaumont <cedric.chaumont@st.com>

Tested-by: Cedric Chaumont <cedric.chaumont@linaro.org> (STM boards)

### CODE REVIEW

File: core/tee/tee_svc_storage.c [hunk 1]
```diff
@@ -978,6 +978,7 @@ TEE_Result tee_svc_storage_start_enum(uint32_t obj_enum, uint32_t storage_id)
 			if (res != TEE_SUCCESS)
 				goto exit;
 			res = tee_obj_verify(sess, o);
+			free(o->pobj->obj_id);
```
Comment (jenswi-linaro) [inline core/tee/tee_svc_storage.c]: The need for this free is a bit obscure, please add a comment explaining why it's needed here.

**Comment by jforissier** [comment]
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey)`

I think you should add 'Fixes #494' to the commit log.

File: core/tee/tee_svc_storage.c (hunk 1 above)
Comment (pascal-brand-st-dev) [inline core/tee/tee_svc_storage.c]: +1

**Comment by cedric-chaumont-st-dev** [comment]
Update commit message.

**Comment by cedric-chaumont-st-dev** [comment]
Update.

**Comment by pascal-brand-st-dev** [comment]
Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

File: core/tee/tee_svc_storage.c [hunk 2]
```diff
@@ -980,6 +980,8 @@ TEE_Result tee_svc_storage_start_enum(uint32_t obj_enum, uint32_t storage_id)
 			res = tee_obj_verify(sess, o);
 			if (res != TEE_SUCCESS)
 				goto exit;
+			free(o->pobj->obj_id);
```
Comment (jenswi-linaro) [inline core/tee/tee_svc_storage.c]: The need for this free is a bit obscure, please add a comment explaining why it's needed here.

**Comment by cedric-chaumont-st-dev** [comment]
@jenswi-linaro : Add comments in the code. Is it OK with you ?

**Comment by jforissier** [comment]
I'm not sure that 'Fix#494' in the commit log will be recognized by GitHub for linking and closing the issue automatically, see https://help.github.com/articles/closing-issues-via-commit-messages/

**Comment by jenswi-linaro** [comment]
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

**Comment by jforissier** [comment]
@cedric-chaumont-st-dev you have modified the patch after I have tested it (you changed the code structure) and still you are keeping my `Tested-by`. You should not do this.

**Comment by cedric-chaumont-st-dev** [comment]
updated

**Comment by jforissier** [comment]
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey)`

**Comment by cedric-chaumont-st-dev** [comment]
Update commit message. Verified (STM board, ARM Juno)
48211 subtests of which 0 failed
701 test cases of which 0 failed
0 test case was skipped
TEE test application done!

**Pull request merged**